src/App/Fossa/Configuration.hs

```diff
@@ -103,16 +108,27 @@ instance FromJSON ConfigPaths where
 defaultFile :: Path Rel File
 defaultFile = $(mkRelFile ".fossa.yml")
 
-readConfigFile :: (Has ReadFS sig m, Has Diag.Diagnostics sig m) => Path Abs File -> m (Maybe ConfigFile)
+readConfigFile :: (Has (Lift IO) sig m, Has ReadFS sig m, Has Diag.Diagnostics sig m) => Path Abs File -> m (Maybe ConfigFile)
 readConfigFile file = do
   exists <- doesFileExist file
   if not exists
     then pure Nothing
     else do
       readConfig <- readContentsYaml @ConfigFile file
       if configVersion readConfig < 3
-        then pure Nothing
+        then withDefaultLogger SevWarn (logWarn warnMsgForOlderConfig) $> Nothing
         else pure $ Just readConfig
+  where
+    warnMsgForOlderConfig :: Doc ann
+    warnMsgForOlderConfig =
+      vsep
+        [ ""
+        , "Incompatible [.fossa.yml] found!"
+        , "Preferences from incompatible [.fossa.yml] will not be considered!"
+        , "Please make sure [.fossa.yml] follows specifications of:"
+        , "    " <> pretty fossaYmlDocUrl
```

Comment:
[nit] consider rewording this; it's not clear _what_ makes the .fossa.yml incompatible. Maybe something like:

```
Incompatible [.fossa.yml] found! Expecting `version: 3`; found `version: 2`
Documentation for the new config file format can be found here: [url]
```